### PR TITLE
added goimports lint check to .golangci.yml

### PR DESCRIPTION
**Pull request by mahmud2011**
#### Summary
* goimports added in `.golangci.yml` linters list
* several files formatted using `goimports -w` to pass goimports lint check

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16621

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
@mahmud2011 - We also need to add the local prefixes setting to the .golangci.yml file and make the changes.

```
goimports:
   local-prefixes: github.com/mattermost/mattermost-server/v5
```

File: .golangci.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 run:
   timeout: 5m
-  modules-download-mode: vendor
+#  https://github.com/golangci/golangci-lint/issues/1502
```
Comment (agnivade) [inline .golangci.yml:3]: You can use any version before 1.34.0 and it should work.

**Comment by mahmud2011** [comment]
> @mahmud2011 - We also need to add the local prefixes setting to the .golangci.yml file and make the changes.
> 
> ```
> goimports:
>    local-prefixes: github.com/mattermost/mattermost-server/v5
> ```

I thought running `goimports -w` will override this, found out `goimports` has a local flag for this.

**Comment by mahmud2011** [comment]
`goimports` seems to ignore the case below, linter passed it as valid.

```go
import (
	"net/smtp"
	"time"

	gomail "gopkg.in/mail.v2"

	"net/http"

	"github.com/jaytaylor/html2text"

	"github.com/mattermost/mattermost-server/v5/mlog"
	"github.com/mattermost/mattermost-server/v5/model"
)
```

**Comment by agnivade** [comment]
Yes, if you deliberately create a separate group, it won't join them. That's by design.

I think you need to run `make app-layers`, `make store-mocks` and `make store-layers` for the CI to pass.

But great work otherwise! Excited for this to go in

**Comment by agnivade** [comment]
Hey @mahmud2011 - to merge this, we need to make some changes in our enterprise repo too. For that, we need to create a branch in this repo. I have created the same PR with your commits in a branch here https://github.com/mattermost/mattermost-server/pull/16640. Let's follow all activity there.

**Pull request closed without merging**